scale-encode/src/impls/composite.rs

```diff
@@ -140,7 +143,7 @@ where
         // Both the target and source type have to have named fields for us to use
         // names to line them up.
         let is_named = {
-            let is_target_named = fields.iter().any(|f| f.name().is_some());
+            let is_target_named = fields.clone().any(|f| f.name().is_some());
```

Comment:
dq: why is clone needed here i.e, does it consume the iterator?

Reply:
Yeah, we are just cloning the iterator so that we don't consume it here, because we want to iterate over it all again later.

(I believe in the common case this just boils down to something cheap because it just needs to copy the function and underlying iterator pointer in most cases, but it allows people to provide more complex iterators if need be too as long as they can be cloned)